device/celestica/x86_64-cel_e1031-r0/sonic_platform/common.py

```diff
@@ -0,0 +1,283 @@
+
+    def load_json_file(self, path):
+        """
+        Retrieves the json object from json file path
+
+        Returns:
+            A json object
+        """
+        with open(path, 'r') as f:
+            json_data = yaml.safe_load(f)
+
+        return json_data
+
+    def get_config_path(self, config_name):
+        """
+        Retrieves the path to platform api config directory
+
+        Args:
+            config_name: A string containing the name of config file.
+
+        Returns:
+            A string containing the path to json file
+        """
+        return os.path.join(self.DEVICE_PATH, self.platform, self.CONFIG_DIR, config_name) if self.is_host() else os.path.join(self.PMON_PLATFORM_PATH, self.CONFIG_DIR, config_name)
+
+    def get_output(self, index, config, default):
+        """
+        Retrieves the output for each function base on config
+
+        Args:
+            index: An integer containing the index of device.
+            config: A dict object containing the configuration of specified function.
+            default: A string containing the default output of specified function.
+
+        Returns:
+            A string containing the output of specified function in config
+        """
+        output_source = config.get('output_source')
+
+        if output_source == self.OUTPUT_SOURCE_IPMI:
+            output = self._ipmi_get(index, config)
+
+        elif output_source == self.OUTPUT_SOURCE_GIVEN_VALUE:
+            output = config["value"]
+
+        elif output_source == self.OUTPUT_SOURCE_GIVEN_CLASS:
+            output = self._get_class(config)
+
+        elif output_source == self.OUTPUT_SOURCE_GIVEN_LIST:
+            output = config["value_list"][index]
+
+        elif output_source == self.OUTPUT_SOURCE_SYSFS:
+            output = self._sysfs_read(index, config)
+
+        elif output_source == self.OUTPUT_SOURCE_FUNC:
+            func_conf = self._main_conf[config['function'][index]]
+            output = self.get_output(index, func_conf, default)
+
+        elif output_source == self.OUTPUT_SOURCE_GIVEN_TXT_FILE:
+            path = config.get('path')
+            output = self.read_txt_file(path)
+
+        elif output_source == self.OUTPUT_SOURCE_GIVEN_VER_HEX_FILE:
+            path = config.get('path')
+            hex_ver = self.read_txt_file(path)
+            output = self._hex_ver_decode(
+                hex_ver, config['num_of_bits'], config['num_of_points'])
+
+        elif output_source == self.OUTPUT_SOURCE_GIVEN_VER_HEX_ADDR:
+            path = config.get('path')
+            addr = config.get('reg_addr')
+            hex_ver = self.get_reg(path, addr)
+            output = self._hex_ver_decode(
+                hex_ver, config['num_of_bits'], config['num_of_points'])
+
+        else:
+            output = default
+
+        return self._clean_output(index, output, config) or default
+
+    def set_output(self, index, input, config):

```

Comment:
>set_output [](http://example.com/codeflow?start=8&length=10)

Seems this function is not used. Please remove.
@mudsut4ke Could you help review?